### DISCUSSION TITLE
EACCES: permission denied,

### DISCUSSION THREAD

**Discussion by abkiwi9** [requarks/wiki#3389] [Help / Questions]
**Question**
Describe in details what you are trying to achieve and where do you need guidance.

**Host Info (please complete the following information):**
OS: [e.g. Ubuntu 18.04, Docker]
Wiki.js version: [e.g. 2.0.1]
Database engine: [e.g. postgres 9.7, mysql 5.7]

**Comment by abkiwi9** [comment]
Hi, I have enabled local file storage as the backup and created a folder /home/wiki-backups added the path, but every time I go and edit a page I get this message: EACCES: permission denied, mkdir '/home/wiki-backups'  folder has permission 755.

Have I set this right?

Regards
Aaron

**Comment by winkelement** [comment]
It would be helpful if you provide more information, for example see **Host Info** above, i guess that template is there for a reason 😃

**Comment by abkiwi9** [comment]
OS: uBUNTU 18.04
Installed your docker version
Wiki.js version: 2.4.107
Database engine: PostgreSQL 11.8
Node.js 12.18.0

So sorry, I got interrupted and totally forgot, my mistake.

**Comment by NGPixel** [comment]
Is that folder writable by the node process?

**Comment by abkiwi9** [comment]
This is the first time I have used node so no.

Do you know where I can find instructions on how to do this?  I have taken a 1-hour look on the internet and can't seem to find anything that looks right to do this correctly. 

Thanks

**Comment by mariyabytes** [comment]
> Do you know where I can find instructions on how to do this? 

Try these steps, in order. 

* `sudo chmod 777 -R /home/wiki-backups` - to allow unfettered access to the folder (temporarily).
* Make a backup. It shouldn't error out. 
* Now  run the command 
`sudo stat /home/wiki-backups | grep "Uid.*" -o `
* This will give you the owner details of the folder. 

```
Uid: ( 1000/  magnum)   Gid: (  985/   users)
```
Here, the user is `magnum` and the group is `users`. 

* Change the folder permission back to safe ones. 

```bash
sudo chown -R user:group /home/wiki-backups
sudo chmod 755 -R /home/wiki-backups
```

This should fix it.

**Comment by abkiwi9** [comment]
Thank you so much :)
I love the wiki.js but have struggled with this software, but getting to know it more every day.

Regards
Aaron

**Comment by abkiwi9** [comment]
I tried it, but it did not work, I have tried every other way I know but I think it is more special as stated above (writable by the node process).

**Comment by winkelement** [comment]
So you created `/home/wiki-backups` locally? Where did you mount it in the container?

**Comment by abkiwi9** [comment]
Yes locally, I think I am getting lost and confused how docker, containers and node work, these are all new to me.

**Comment by winkelement** [comment]
It might make sense for you to join the [Slack Workspace](https://wiki.requarks.io/slack) and try to get help there. 
I remember there was a thread about that already https://requarks.slack.com/archives/C49J7AB8R/p1580618622153600

Once you (hopefully) solve it you can report back your solution here for others to find it when they search the issues.
